cdap-ui/app/cdap/components/PipelineDetails/PipelineDetailsTopPanel/PipelineDetailsButtons/PipelineRunButton.js

```diff
@@ -48,18 +48,17 @@ export default class PipelineRunButton extends Component {
     });
   };
 
-  runPipelineOrToggleConfig = () => {
+  onRunPipelineClick = () => {
     if (this.props.runButtonLoading) {
       return;
     }
 
     if (!this.state.showRunOptions) {
-      let { isMissingKeyValues } = PipelineConfigurationsStore.getState();
-      if (isMissingKeyValues) {
-        this.toggleRunConfigOption();
-      } else {
-        runPipeline();
-      }
+      let { runtimeArgs } = this.props;
+      // Arguments with empty values are assumed to be provided from the pipeline
+      runtimeArgs.pairs = runtimeArgs.pairs.filter((runtimeArg) => !runtimeArg.value);
+      let runtimeArgsMap = convertKeyValuePairsToMap(runtimeArgs.pairs);
+      runPipeline(runtimeArgsMap);
```

Comment:
don't we still want the runtime argument modal to actually open? I think we should still open the modal but not force the user to fill everything out.

Reply:
So, we'd open the modal when user clicks `run` with empty arguments and they'd have to click `run` again to actually run the pipeline? Isn't that a bit confusing? I'll check with Lea and make these changes

Reply:
Talked to Lea about this, leaving as is, when there are some runtime arguments are missing, we will show the popover. Will execute the pipeline if none are missing.